PostHog/PostHogContext.swift

```diff
@@ -56,28 +56,62 @@ class PostHogContext {
         #if os(iOS) || os(tvOS)
             let device = UIDevice.current
             // use https://github.com/devicekit/DeviceKit
+            let processInfo = ProcessInfo.processInfo
+
             properties["$device_name"] = device.model
-            properties["$os_name"] = device.systemName
-            properties["$os_version"] = device.systemVersion
-
-            var deviceType: String?
-            switch device.userInterfaceIdiom {
-            case UIUserInterfaceIdiom.phone:
-                deviceType = "Mobile"
-            case UIUserInterfaceIdiom.pad:
-                deviceType = "Tablet"
-            case UIUserInterfaceIdiom.tv:
-                deviceType = "TV"
-            case UIUserInterfaceIdiom.carPlay:
-                deviceType = "CarPlay"
-            case UIUserInterfaceIdiom.mac:
-                deviceType = "Desktop"
-            default:
-                deviceType = nil
-            }
-            if deviceType != nil {
-                properties["$device_type"] = deviceType
+
+            // iOS app running in compatibility mode (Designed for iPad/iPhone)
+            var isiOSAppOnMac = false
+
+            #if os(iOS)
+                if #available(iOS 14.0, *) {
+                    isiOSAppOnMac = processInfo.isiOSAppOnMac
+                }
+            #endif
+
+            // iPad app running on Mac Catalyst
+            #if targetEnvironment(macCatalyst)
+                let isMacCatalystApp = true
+            #else
+                let isMacCatalystApp = false
+            #endif
+
+            if isMacCatalystApp || isiOSAppOnMac {
+                let underlyingOS = device.systemName
+                let underlyingOSVersion = device.systemVersion
+                let macOSVersion = processInfo.operatingSystemVersionString
+                // e.g macOS Version 15.1.1 (Build 24B91) / iPadOS 18.1
+                properties["$os_version"] = "macOS \(macOSVersion) / \(underlyingOS) \(underlyingOSVersion)"
+                properties["$os_name"] = isMacCatalystApp
+                    ? "macOS (Mac Catalyst)"
+                    : "macOS (iOS running on Mac)"
```

Comment:
The fix is great but we need an alternative solution here.
People use the os_name and os_version with matching values in feature flags, hogql, etc.
Maybe we need a new property to tell if it's a mac catalyst or not then? like a boolean one.

Reply:
I think whether it's catalyst or not is more about the app itself and not the OS, so maybe this info should be with one of the `$app_X` properties. the underlying OS is still the same.

Reply:
I think `$os_` should be about the running OS on the physical device, if that's wrong, we have to fix it.
I think `$device_` should be about the physical device, if that's wrong, we have to fix it.
Adding a new app flag makes sense.

Reply:
> For iOS running on Mac, $os_ points to iPadOS

@ioannisj not sure if this is correct? its a mac, so the OS is the mac os, will people think they are running the app on an ipadoS?

Reply:
So again just to confirm, the `os` and `device` props should be related to the physical device.
if the app is in compat. mode, this info should go either with `app` props or new props.
if you want to add the info of the emulated OS in compat mode, this should go under new props, right now `device` and `os` props are meant for the physical device.